CMake/RealmCore.cmake

```diff
@@ -238,25 +237,41 @@ endfunction()
 
 macro(build_realm_core)
     set(core_prefix_directory "${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/realm-core")
-    ExternalProject_Add(realm-core
-        PREFIX ${core_prefix_directory}
-        BUILD_IN_SOURCE 1
-        BUILD_COMMAND make -C src/realm librealm.a librealm-dbg.a ${MAKE_FLAGS}
-        INSTALL_COMMAND ""
-        CONFIGURE_COMMAND ""
-        ${USES_TERMINAL_BUILD}
-        ${ARGV}
-        )
+    if(REALM_PLATFORM STREQUAL "Android")
+      ExternalProject_Add(realm-core
+            PREFIX ${core_prefix_directory}
+            BUILD_IN_SOURCE 1
+            BUILD_COMMAND sh build.sh build-android
```

Comment:
Since this command should be the only difference, it'd be better to hoist it into a variable and leave only a single call to `ExternalProject_add`.